File: app/src/main/java/com/electricdreams/shellshock/feature/settings/SettingsActivity.java
package com.electricdreams.shellshock.feature.settings;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.electricdreams.shellshock.R;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        
        // Setup back button
        View backButton = findViewById(R.id.back_button);
        if (backButton != null) {
            backButton.setOnClickListener(v -> finish());
        }
        
        // Setup navigation to sub-settings
        findViewById(R.id.currency_settings_item).setOnClickListener(v -> {
            startActivity(new Intent(this, CurrencySettingsActivity.class));
        });
        
        findViewById(R.id.mints_settings_item).setOnClickListener(v -> {
            startActivity(new Intent(this, MintsSettingsActivity.class));
        });
        
        findViewById(R.id.items_settings_item).setOnClickListener(v -> {
            startActivity(new Intent(this, ItemsSettingsActivity.class));
        });
    }
}

File: app/src/main/java/com/electricdreams/shellshock/feature/settings/SettingsActivity.kt
package com.electricdreams.shellshock.feature.settings

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.electricdreams.shellshock.R

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        findViewById<View?>(R.id.back_button)?.setOnClickListener { finish() }

        findViewById<View>(R.id.theme_settings_item).setOnClickListener {
            startActivity(Intent(this, ThemeSettingsActivity::class.java))
        }

        findViewById<View>(R.id.currency_settings_item).setOnClickListener {
            startActivity(Intent(this, CurrencySettingsActivity::class.java))
        }

        findViewById<View>(R.id.mints_settings_item).setOnClickListener {
            startActivity(Intent(this, MintsSettingsActivity::class.java))
        }

        findViewById<View>(R.id.items_settings_item).setOnClickListener {
            startActivity(Intent(this, ItemsSettingsActivity::class.java))
        }

        findViewById<View>(R.id.security_settings_item).setOnClickListener {
            startActivity(Intent(this, SecuritySettingsActivity::class.java))
        }
    }
}
